### DISCUSSION TITLE
How to utilize renames.txt?

### DISCUSSION THREAD

**Discussion by torbjornsk** [newren/git-filter-repo#196] [Q&A] [answered]
I'm trying to use filter-repo to split a repository into multiple. I'd like to retain the history also when a file is moved/renamed, and it seems that the analyse mode provides information on this through renames.txt. But, I can't seem to find any information as to how this file can be fed to filter-repo as paths to keep. How is this handled?

**Comment by newren** [comment, marked as answer]
renames.txt uses a number of heuristics (i.e. it can get things wrong especially in really weird cases), so it's there merely to guide you on picking pathnames you might want to manually pass to filter-repo.
